### PR TITLE
Fix missing constr import issue

### PR DESCRIPTION
**Pull request by koxudaxi**
Closes: https://github.com/koxudaxi/datamodel-code-generator/issues/1495

### CODE REVIEW

**Comment by koxudaxi** [comment]
@indrat 
Could you please review it? if you have the time.

File: tests/data/expected/main/main_openapi_const_field_msgspec/output.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 from __future__ import annotations
 
-from msgspec import Struct
+from msgspec import Meta, Struct
```
Comment (koxudaxi) [inline tests/data/expected/main/main_openapi_const_field_msgspec/output.py:7]: @indrat 
I think the PR fixes the issue too https://github.com/koxudaxi/datamodel-code-generator/issues/1607

**Comment by indrat** [comment]
Testing this with the msgspec output and `--target-python 3.8` has an issue with the isort phase where the generated code for `common.py` has a `from typing_extensions import` but doesn't import anything:

```python
from __future__ import annotations
from typing import Any
from typing_extensions import 

Model = Any
```
and thus the isort call fails with:
```bash
› datamodel-codegen --input root_test --input-file-type jsonschema --output output_test --disable-timestamp --enable-version-header --collapse-root-models --output-model-type msgspec.Struct --target-python 3.8
Traceback (most recent call last):
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/datamodel_code_generator/__main__.py", line 388, in main
    generate(
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/datamodel_code_generator/__init__.py", line 435, in generate
    results = parser.parse()
              ^^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/datamodel_code_generator/parser/base.py", line 1183, in parse
    body = code_formatter.format_code(body)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/datamodel_code_generator/format.py", line 164, in format_code
    code = self.apply_isort(code)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/datamodel_code_generator/format.py", line 193, in apply_isort
    return isort.code(code, config=self.isort_config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/venv/lib/python3.11/site-packages/isort/api.py", line 92, in sort_code_string
    sort_stream(
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/venv/lib/python3.11/site-packages/isort/api.py", line 210, in sort_stream
    changed = core.process(
              ^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/venv/lib/python3.11/site-packages/isort/core.py", line 422, in process
    parsed_content = parse.file_contents(import_section, config=config)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/indra/projects/data-model-code-gen/datamodel-code-generator/venv/lib/python3.11/site-packages/isort/parse.py", line 522, in file_contents
    if "," in import_string.split(just_imports[-1])[-1]:
                                  ~~~~~~~~~~~~^^^^
IndexError: list index out of range
```

File: tests/data/expected/main/main_openapi_const_field_msgspec/output.py (hunk 1 above)
Comment (indrat) [reply tests/data/expected/main/main_openapi_const_field_msgspec/output.py:7]: This does fix the missing `constr` import and does appear to fix a similar issue for the msgspec output but not #1607 specifically. #1607 *isn't* using `--collapse-root-models` so is ending up with the root model in `common.py` without the `Meta` import.

Comment (koxudaxi) [reply tests/data/expected/main/main_openapi_const_field_msgspec/output.py:7]: Thank you for checking the pattern.
OK, I will fix the issue in another PR.

File: tests/test_main.py [hunk 2]
```diff
@@ -5804,3 +5804,59 @@ def test_main_msgspec_use_annotated_with_field_constraints():
...
+                str(JSON_SCHEMA_DATA_PATH / 'duplicate_field_constraints'),
+                '--output',
+                str(output_path),
+                '--input-file-type',
+                'jsonschema',
+                '--collapse-root-models',
+                '--output-model-type',
+                'msgspec.Struct',
+                '--target-python-version',
+                '3.8',
```
Comment (koxudaxi) [inline tests/test_main.py:5851]: @indrat 
I have added the test case.

**Comment by indrat** [comment]
lgtm 👍

**Review by indrat — APPROVED** [review]

**Comment by koxudaxi** [comment]
Thank you very much!!

**Pull request merged**